designcompose/src/main/java/com/android/designcompose/squoosh/SquooshAnimate.kt

```diff
@@ -74,11 +93,30 @@ internal class SquooshAnimatedLayout(
             return
         }
 
+        val fromWidth = if (hasRotation) from.view.style.fixedWidth(density) else fromLayout.width
+        val fromHeight =
+            if (hasRotation) from.view.style.fixedHeight(density) else fromLayout.height
+        val toWidth = if (hasRotation) to.view.style.fixedWidth(density) else toLayout.width
+        val toHeight = if (hasRotation) to.view.style.fixedHeight(density) else toLayout.height
+
+        if (hasRotation) {
+            // Interpolate the transformation matrix
+            val transform: MutableList<Float> = MutableList(16) { 0F }
+            for (i in 0..15) {
+                transform[i] = toTransform[i] * value + fromTransform[i] * iv
```

Comment:
I added some code to extract the skew values out of the transform matrices and they look correct. However, if I take the rotation, scale, translate and skew values and construct a matrix out of it with the formula:
matrix = translation * rotation * scale * skew, the resulting matrix seems to be missing some information from the original, so that the skew animation pops the skewed rectangle to be taller in the Y axis during the animation. It's possible there's a bug in the decomposition function, or I have a bug in reconstruction of the matrix. I don't know matrices enough to know what might go wrong with a simple interpolation of all the values, but in my tests so far that is what looks the best.

Reply:
@iamralpht let me know if you have more thoughts on interpolating the transforms vs decomposition/recomposition. Then I'll clean up the code a bit.

Reply:
Ok, do you want to put this in as-is and make a ticket to fix the decomposition? I linked some other examples in the other comment, so perhaps there's something there you didn't already try, otherwise the examples look good.

Reply:
Fixed decomposition.